game.py

```diff
@@ -83,72 +87,79 @@ def main(self):
         while True:
             self.draw_grid(False)
             if self.move_made:
-                future_square_cord_color = self.draw_predicted(next_colors)
-                self.future_square_cord_color = future_square_cord_color
-                for x_grid, y_grid, color in self.future_square_cord_color:
-                    lines = self.find_adjacent_color((x_grid, y_grid), color)
-                    self.check_length_remove_square(lines)
+                try:
+                    success = self.process_turn(next_colors)
+                    if not success:
+                        print('Failed to process turn.')
+                except Exception as e:
+                    print(f'An error occurred: {e}')
+                self.spawn = True
                 next_colors = [rand_color() for _ in range(3)]
                 self.draw_future_grid(next_colors)
                 self.move_made = False
-
             self.handle_mouse_click()
             self.handle_selected_square()
-            self.score()
-            self.movesmade()
+            STATS.score(self)
+            STATS.movesmade(self)
             pygame.display.update()
 
+    def select_square(self, x, y):
+        x, y = normalize_cords(x, y)
+        x_grid = int((x / 50) - 3)
+        y_grid = int((y / 50) - 3)
+        return x_grid, y_grid
+
+    def square_path(self, start, end):
+        path = astar(self.space, start, end)
+        first = path[0]
+        last = path[-1]
+        color = self.sqr_grid[first[0]][first[1]].color
+        self.space[last[0]][last[1]] = 1
+        self.space[first[0]][first[1]] = 0
+        self.move_square(path, color)
+        lines = self.find_adjacent_color(last, color)
+        self.check_length_remove_square(lines)
+        self.move_made = True
+        return path, color
+
     def handle_mouse_click(self):
         for event in pygame.event.get():
             if event.type == pygame.MOUSEBUTTONUP:
                 x, y = pygame.mouse.get_pos()
+                x_grid, y_grid = self.select_square(x, y)
                 if x < OFFSET or y < OFFSET:
                     break
-                x, y = normalize_cords(x, y)
-                x_grid = int((x / 50) - 3)
-                y_grid = int((y / 50) - 3)
                 if self.selected_square and self.space[x_grid][y_grid] == 0:
                     start = (
                         self.selected_square.grid_x,
                         self.selected_square.grid_y,
                     )
                     end = (x_grid, y_grid)
-                    path = astar(self.space, start, end)
+                    path = self.square_path(start, end)
```

Comment:
what does square_path do? the name is unclear.
Why does it return a path? Specially after the square is moved. What good does that do?